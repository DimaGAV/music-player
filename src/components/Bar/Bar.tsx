import Controls from "@/components/Controls/Controls";
import TrackPlay from "@/components/TrackPlay/TrackPlay";
import Volume from "@/components/Volume/Volume";
import styles from "./Bar.module.css"

const Bar = () => {
    return ( <div className={styles.bar}>
        <div className={styles.barContent}>
          <div className={styles.barPlayerProgress} />
          <div className={styles.barPlayerBlock}>
            <div className={styles.barPlayer}>
              <Controls/>
              <TrackPlay/>
            </div>
            <Volume/>
          </div>
        </div>
      </div> );
}
 
export default Bar;