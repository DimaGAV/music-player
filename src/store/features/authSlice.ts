import { singInUser } from "@/api/playlist";
import { createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";

export const getUser = createAsyncThunk(
  "user/getUser",
  async (userData: {
    email: string,
    password: string
  }) => { 
    const user = await singInUser(userData)
    return user
  }
)

type AuthStateType =  {
  authState: boolean;
}

const initialState: AuthStateType = {
  authState: false,
};

const authSlice = createSlice({
  name: "auth",
  initialState,
  reducers: {
    setAuthState: (state, action: PayloadAction<boolean>) => {
      state.authState = action.payload;
    },
  },
});

export const { setAuthState } = authSlice.actions;
export const authReducer = authSlice.reducer;