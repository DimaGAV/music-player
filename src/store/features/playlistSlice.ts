import { PlaylistType } from "@/types/playlist";
import { createSlice, PayloadAction } from "@reduxjs/toolkit";

type PlaylistStateType =  {
  currentTrack: null | PlaylistType;
  playlist: PlaylistType[]
}

const initialState: PlaylistStateType = {
  currentTrack: null,
  playlist: []
};

const playlistSlice = createSlice({
  name: "playlist",
  initialState,
  reducers: {
    setCurrentTrack: (state, action: PayloadAction<PlaylistType>) => {
      state.currentTrack = action.payload;
    },
  },
});

export const { setCurrentTrack } = playlistSlice.actions;
export const playlistReducer = playlistSlice.reducer;