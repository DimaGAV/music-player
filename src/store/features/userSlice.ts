import { singInUser } from "@/api/playlist";
import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";

export const getUser = createAsyncThunk(
  "user/getUser",
  async (userData: { email: string; password: string }) => {
    const user = await singInUser(userData);
    return user;
  }
);
//создать getToken
const userSlice = createSlice({
  name: "user",
  initialState: {
    user: null,
  },
  reducers: {
    logout: (state) => {
      state.user = null;
    },
  },
  extraReducers: (builder) => {
    // Обрабатываем экшены, связанные с нашим асинхронным thunk
    builder
      // Обработка успешного выполнения асинхронного экшена getUser
      .addCase(getUser.fulfilled, (state, action) => {
        state.user = action.payload; // Обновляем состояние пользователя данными из action.payload
      })
      // Обработка неудачного выполнения асинхронного экшена getUser
      .addCase(getUser.rejected, (state, action) => {
        console.error("Error:", action.error.message); // Выводим сообщение об ошибке в консоль
      });
    // .addCase
  },
});

export const { logout } = userSlice.actions;
export const userReducer = userSlice.reducer;
