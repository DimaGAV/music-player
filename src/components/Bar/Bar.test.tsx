import renderer from "react-test-renderer";
import Bar from "./Bar";
import ReduxProvider from "@/store/ReduxProvider";
import { PlayerStateProvider } from "@/contexts/PlayerStateContext";
import { PlaylistStateType } from "@/store/features/playlistSlice";

it("renders correctly", () => {

  const initialState: PlaylistStateType = {
currentTrack: {
    _id: 2,
    name: "Test Track 2",
    author: "Test Artist 2",
    release_date: "2024-02-01",
    genre: "Rock",
    duration_in_seconds: 210,
    album: "Test Album 2",
    logo: "/path/to/logo2.png",
    track_file: "/path/to/track2.mp3",
    stared_user: [],
}
  }

  const tree = renderer
    .create(
      <ReduxProvider>
        <PlayerStateProvider>
          <Bar />
        </PlayerStateProvider>
      </ReduxProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
