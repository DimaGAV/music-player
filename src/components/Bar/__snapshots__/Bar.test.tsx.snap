// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bar"
>
  <div
    className="barContent"
  >
    <div
      className="barTimer"
    >
      00:00
       / 
      00:00
    </div>
    <audio
      onTimeUpdate={[Function]}
      src="/path/to/track2.mp3"
    />
    <input
      className="styledProgressInput"
      max={0}
      min={0}
      onChange={[Function]}
      step={0.01}
      type="range"
      value={0}
    />
    <div
      className="barPlayerBlock"
    >
      <div
        className="barPlayer"
      >
        <div
          className="playerControls"
        >
          <div
            className="playerBtnPrev"
            onClick={[Function]}
          >
            <svg
              className="playerBtnPrevSvg"
            >
              <use
                xlinkHref="/img/icon/sprite.svg#icon-prev"
              />
            </svg>
          </div>
          <div
            className="playerBtnPlay"
            onClick={[Function]}
          >
            <svg
              className="playerBtnPlaySvg"
            >
              <use
                xlinkHref="/img/icon/sprite.svg#icon-play"
              />
            </svg>
          </div>
          <div
            className="playerBtnNext"
            onClick={[Function]}
          >
            <svg
              className="playerBtnNextSvg"
            >
              <use
                xlinkHref="/img/icon/sprite.svg#icon-next"
              />
            </svg>
          </div>
          <div
            className="playerBtnRepeat"
            onClick={[Function]}
          >
            <svg
              className="playerBtnRepeatSvg"
            >
              <use
                xlinkHref="/img/icon/sprite.svg#icon-repeat"
              />
            </svg>
          </div>
          <div
            className="playerBtnShuffle"
            onClick={[Function]}
          >
            <svg
              className="playerBtnShuffleSvg"
            >
              <use
                xlinkHref="/img/icon/sprite.svg#icon-shuffle"
              />
            </svg>
          </div>
        </div>
        <div
          className="playerTrackPlay"
        >
          <div
            className="trackPlayContain"
          >
            <div
              className="trackPlayImage"
            >
              <svg
                className="trackPlaySvg"
              >
                <use
                  xlinkHref="/img/icon/sprite.svg#icon-note"
                />
              </svg>
            </div>
            <div
              className="trackPlayAuthor"
            >
              <a
                className="trackPlayAuthorLink"
                href="http://"
              >
                Test Artist 2
              </a>
            </div>
            <div
              className="trackPlayAlbum"
            >
              <a
                className="trackPlayAlbumLink"
                href="http://"
              >
                Test Album 2
              </a>
            </div>
          </div>
          <div
            className="trackPlayLikeDis"
          >
            <div
              className="trackPlayLike"
              onClick={[Function]}
            >
              <svg
                className="trackPlayLikeSvg"
              >
                <use
                  xlinkHref="/img/icon/sprite.svg#icon-dislike"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
